### PR TITLE
Big refactor to organise metadata by API group rather than per-API

### PR DESCRIPTION
**Pull request by bloodearnest**

### CODE REVIEW

**Comment by bloodearnest** [comment]
Preview of some docs built with this branch is here: https://private-fileshare.canonical.com/~wavy/snapdevicegw/en/metadata.html

File: acceptable/__main__.py [hunk 1]
```diff
@@ -264,53 +225,39 @@ def render_markdown(metadata, cli_args):
...
+        all_undocumented = all(
+            api.get('undocumented', False) for api in apis.values()
         )
-    else:
-        default_group = api_groups.pop(None, None)
-        if default_group is not None:
-            navigation.extend(
-                sorted(default_group, key=sort_key),
+
+        if not all_undocumented:
```
Comment (sparkiegeek) [inline acceptable/__main__.py]: Instead of checking for `not all undocumented` we could do `any documented` instead

File: acceptable/__main__.py [hunk 2]
```diff
@@ -264,53 +225,39 @@ def render_markdown(metadata, cli_args):
...
+            page_file = '{}.{}'.format(group, cli_args.extension)
+            page = {'title': group.title(), 'location': page_file}
+            path = os.path.join('en', page_file)
+            navigation.append(page)
```
Comment (sparkiegeek) [inline acceptable/__main__.py]: can you move these lines around, so e.g. `navigation.append(page)` is after making `page` and the `path` is closer to where it's used too (in the `yield`)

It's hard to follow when they're tangled up like this IMHO

**Comment by sparkiegeek** [comment]
[Acceptable: page-per api group](https://trello.com/c/a5Lkq7rO/209-acceptable-page-per-api-group)
